core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -543,7 +560,14 @@ class ReplicaManager(val config: KafkaConfig,
           val partitionOpt = getPartition(topicPartition)
           val info = partitionOpt match {
             case Some(partition) =>
-              partition.appendRecordsToLeader(records, isFromClient, requiredAcks)
+              try {
+                partition.appendRecordsToLeader(records, isFromClient, requiredAcks)
+              } catch {
+                case e: KafkaStorageException =>
```

Comment:
It seems that sometimes we are catching both IOException and KafkaStorageException, sometimes we only catch  KafkaStorageException. It might be better to only catch IOException at the the direct thrown and convert them to KafkaStorageException. So in the caller methods we only need to catch KafkaStorageException.